From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cpw <cpw@weeksfamily.ca>
Date: Sun, 2 Aug 2015 20:56:16 -0400
Subject: [PATCH] Merge any LVTT to an incoming LVT (if they're out of
 sequence), and merge LVTTs together if there's no LVT yet.


diff --git a/src/org/jetbrains/java/decompiler/struct/StructMember.java b/src/org/jetbrains/java/decompiler/struct/StructMember.java
index 724c0db23d607d1e243f8cf278b0a61a9789e19d..af1df05e4120b1053ad0d1e74da88810c3b97487 100644
--- a/src/org/jetbrains/java/decompiler/struct/StructMember.java
+++ b/src/org/jetbrains/java/decompiler/struct/StructMember.java
@@ -62,6 +62,15 @@ public class StructMember {
           StructLocalVariableTableAttribute table = (StructLocalVariableTableAttribute)attributes.getWithKey(StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TABLE);
           table.addLocalVariableTable((StructLocalVariableTableAttribute)attribute);
         }
+        else if (StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TABLE.equals(name) && attributes.containsKey(StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TYPE_TABLE)) {
+            StructLocalVariableTableAttribute lvtt = (StructLocalVariableTableAttribute)attributes.getWithKey(StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TYPE_TABLE);
+            ((StructLocalVariableTableAttribute)attribute).addLocalVariableTable(lvtt);
+            attributes.addWithKey(attribute, attribute.getName());
+        }
+        else if (StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TYPE_TABLE.equals(name) && attributes.containsKey(StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TYPE_TABLE)) {
+            StructLocalVariableTableAttribute table = (StructLocalVariableTableAttribute)attributes.getWithKey(StructGeneralAttribute.ATTRIBUTE_LOCAL_VARIABLE_TYPE_TABLE);
+            table.addLocalVariableTable((StructLocalVariableTableAttribute)attribute);
+        }
         else {
           attributes.addWithKey(attribute, attribute.getName());
         }
